aten/src/ATen/native/Copy.cpp

```diff
@@ -38,8 +38,8 @@ namespace at {
 namespace native {
 
 Tensor& _s_copy__cpu(Tensor& self, const Tensor& src, bool non_blocking) {
-  if (src.is_cuda()) {
-    _s_copy_from(src, self, non_blocking);
+  if (src.is_cuda() || src.is_xla()) {
```

Comment:
this is wrong (although it was wrong before, not your fault).  The "cpu" here is referring to the backend, not the device type, so this should be looking at the CPU Tensor Type Id (i'll submit a PR for that).

Reply:
Sure, if you fix that part, I can fix the _s_copy_from one.
Or I am not offended at all if you want to fix that as well :)